app/frontend/src/features/userQueries/useUsers.ts

```diff
@@ -0,0 +1,54 @@
+import { Error } from "grpc-web";
+import { useEffect } from "react";
+import { useQueries, useQueryClient } from "react-query";
+import { User } from "../../pb/api_pb";
+import { service } from "../../service";
+import { userStaleTime } from "./constants";
+
+export default function useUsers(ids: number[], invalidate: boolean = false) {
+  const queryClient = useQueryClient();
+  useEffect(() => {
+    if (invalidate) {
+      queryClient.invalidateQueries({
+        predicate: (query) =>
+          query.queryKey[0] === "user" &&
+          ids.indexOf(query.queryKey[1] as number) > -1,
+      });
+    }
+  }, [invalidate, queryClient, ids]);
+
+  const queries = useQueries<User.AsObject, Error>(
+    ids.map((id) => ({
+      queryKey: ["user", id],
+      queryFn: () => service.user.getUser(id.toString()),
+      staleTime: userStaleTime,
+    }))
+  );
+
+  const errors = queries
+    .map((query) => query.error?.message)
+    .filter((e): e is string => typeof e === "string");
+  const isLoading = queries.some((query) => query.isLoading);
+  const isError = !errors.length;
```

Comment:
```suggestion
  const isError = !!errors.length;
```
Shouldn't it be this instead? If the errors array is empty, `!errors.length = !0 = true` but it should be `false` since there aren't any errors.

Reply:
This is why JS is silly 😤